modules/admin-ui/src/test/resources/karma.conf.js

```diff
@@ -52,14 +52,41 @@ module.exports = function (config) {
 
         autoWatch : true,
 
-        frameworks: ['jasmine'],
+        frameworks: ['detectBrowsers', 'jasmine'],
 
-        browsers : ['PhantomJS'],
+        detectBrowsers: {
+            enabled: true,
+            usePhantomJS: true,
```

Comment:
Since [PhantomJS is long since abanfoned](https://github.com/ariya/phantomjs/issues/15344), do we want to continue using it? For a lot of the other test dependencies (ffprobe, sox, synfig, …) we just don't run the tests if the dependency is not present and we could do that here as well. That would also make the build system a bit simpler since we don't need to check for special cases and would make things way easier on systems like Alpine Linux.

That said, I don't have a very stromg opinion here and if you want to keep it, I won't start a fight ;-P

Reply:
What does removing Phantom do for Alpine?

I don't really have a strong feeling about this either way, but here's my reasoning:

If, somehow, Opencast is being built on a system without a browser installed we want to be able to at least have some sense of whether things are working as expected.  That's where Phantom comes in.  It's not ideal, which is also why it's excluded if anything better is available, but it allows automated build images built without a browser to continue working.

That being said, it's not that big a deal to include Firefox as a build dep.

Reply:
> What does removing Phantom do for Alpine?

By default, Opencast does not build on systems like Apine since Opencast downloads the PhantomJS binary which requires the glibc. Since most Linux distributions use glibc, Opencast builds on most Linux systems. Unfortunately, most are not all. Apline, for instance, uses the [musl libc](https://musl.libc.org/) since it is quite a bit smaller. So, building Opencast on such systems is very hard due to parts like PhantomJS.

> If, somehow, Opencast is being built on a system without a browser installed we want to be able to at least have some sense of whether things are working as expected.

The recent discussion about build systems at the Dev'n'Hack camp made me realize that these tests mainly have two target audiences:

1. **CI systems** which, by now, are certainly our main defense against accidentally pushing broken stuff into the project – and the CI systems only work because of the tests. Everyone has to pass these unless it's local code which not a lot of people still have and which is their own responsibility.

2. The team of **core developers** needs a local testing environment since they are making a lot of changes to the project.

In both cases, having a browser isn't hard. In case of a new developer not having a browser on their machine – unlikely but could happen – this does not really matter since worst case, patches would fail in the CI.

Finally, looking at users, no tests do not matter at all for several reasons:

- Hardly any users build their own code today. They use packages, containers, tarballs, … meaning that they do not run these tests.
- People do not build on the systems they deploy on. We discourage that… which makes sense.
- If tests for release versions fail for someone, we usually advise them to disable tests. After all, we know that the tests have passed on our systems so it's likely a glitch in the tests.

That is why I tend to go with simpler is better, preventing us from having to find out that things failed for someone just because they did not have r.g. Firefox installed and we didn't realize that this failed on a certain system since we are testing with browsers.

Reasons explained, I still leave the final to you ;-)

Reply:
Fair enough.  This will probably break Buildbot, and possibly Travis.  Let's see.